src/deferred.js

```diff
@@ -316,14 +314,17 @@ jQuery.extend( {
 					if ( values === progressValues ) {
 						master.notifyWith( contexts, values );
 					} else if ( !( --remaining ) ) {
-						master.resolveWith( contexts, values );
+						master.resolveWith(
+							contexts.length === 1 ? contexts[ 0 ] : contexts,
+							values
```

Comment:
I don't like the incongruity of popping out the context but not the value. And looking at the discussion in #2018 around casting and idempotence, I'm almost certain that both should be unwrapped when `contexts.length === 1`.